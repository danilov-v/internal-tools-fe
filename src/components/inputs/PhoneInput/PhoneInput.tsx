import React, { useRef } from 'react';
import useMaskedInput from '@viewstools/use-masked-input';
import { PHONE_MASK } from 'helpers/mask';
import { Input, InputProps } from 'components/inputs/Input';

const PhoneInput: React.FC<PhoneInputProps> = ({
  onChange,
  value,
  placeholder,
  guide = false,
  keepCharPositions = false,
  ...props
}) => {
  const input = useRef(null);

  const change = useMaskedInput({
    input,
    guide,
    keepCharPositions,
    showMask: true,
    mask: PHONE_MASK,
    onChange,
    value,
  });

  return (
    <Input
      {...props}
      innerRef={input}
      onChange={change}
      placeholder={placeholder || '(29) 123-12-13'}
    />
  );
};

interface PhoneInputProps extends InputProps {
  guide?: boolean;
  keepCharPositions?: boolean;
}

export { PhoneInput };
