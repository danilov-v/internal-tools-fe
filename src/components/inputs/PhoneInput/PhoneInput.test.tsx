import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { PhoneInput } from './PhoneInput';

import 'jest-styled-components';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    ...props,
  };

  return render(<PhoneInput {...parsedProps} />);
};

describe('PhoneInput component', () => {
  it('matches snapshot with required props only', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });
});
