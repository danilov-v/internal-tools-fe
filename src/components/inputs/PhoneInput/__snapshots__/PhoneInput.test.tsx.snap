// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneInput component Should render and matche snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-tap-highlight-color: transparent;
  background: none;
  border-radius: 2px;
  border-style: solid;
  border-width: 0px;
  border-bottom-width: 1px;
  font-size: 16px;
  margin: 0;
  outline: none;
  text-align: left;
}

.c1:hover,
.c1:focus {
  box-shadow: 0 1px 0;
}

<body>
  <div>
    <div
      class="c0"
    >
      <input
        class="c1"
        color="primary"
        placeholder="(29) 123-12-13"
        type="text"
        value=""
      />
    </div>
  </div>
</body>
`;
